hoodie-client/src/main/java/com/uber/hoodie/io/HoodieAppendHandle.java

```diff
@@ -159,11 +159,14 @@ private void init(String partitionPath) {
     return Optional.empty();
   }
 
+  // TODO (NA) - Perform a schema check of current input record with the last schema on log file
```

Comment:
If schema is backwards compatible, this will not happen. But for the COW case, we fail fast since when merging when we see a smaller schema but in this case it will happily append and move on and fail in compaction.

Reply:
okay. fair enough